### PR TITLE
e2e: add benchmarking functionality

### PR DESCRIPTION
**Pull request by cmwaters**
This PR adds the `benchmark` command to the e2e runner binary. This samples a given testnet over a hundred blocks and prints out:
- The average block interval
- The standard deviation
- The fastest block
- The slowest block

This should be seen as an elementary method of benchmarking a testnet. 

It does not run perbutations.

Sample from `master`:

```
Benchmarked from height 1030 to 1129
        Mean Block Interval: 1.434472893s
        Standard Deviation: 0.127005
        Max Block Interval: 2.311582746s
        Min Block Interval: 1.304716766s
```

### CODE REVIEW

File: test/e2e/runner/benchmark.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	return nil
+}
+
+type testnetStats struct {
+	from int64
+	to   int64
+	mean time.Duration
+	std  float64
+	max  time.Duration
+	min  time.Duration
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:61]:
```suggestion
	startHeight int64
	endHeight   int64
	mean time.Duration
	std  float64
	maxHeight  time.Duration
	minHeight time.Duration
```

File: test/e2e/runner/benchmark.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+		t.std,
+		t.max,
+		t.min,
+	)
+}
+
+// fetchBlockChain fetches a complete, up-to-date block history from
+// the freshest testnet archive node.
+func fetchBlockChainSample(testnet *e2e.Testnet, benchmarkLength int64) ([]*types.BlockMeta, error) {
+
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:83]:
```suggestion
```

File: test/e2e/runner/benchmark.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+)
+
+// Benchmark is a simple function for fetching, calculating and printing
+// the following metrics:
+// 1. Average block production time
+// 2. Block interval standard deviation
+// 3. Max block interval (slowest block)
+// 4. Min block interval (fastest block)
+func Benchmark(testnet *e2e.Testnet, benchmarkLength int64) error {
+
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:20]:
```suggestion
```

File: test/e2e/runner/benchmark.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	// wait for the length of the benchmark period in blocks to pass. We allow 5 seconds for each block
+	// which should be sufficient.
+	waitingTime := time.Duration(benchmarkLength*5) * time.Second
+	endHeight, err := waitForAllNodes(testnet, block.Height+benchmarkLength, waitingTime)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Ending benchmark period.", "height", endHeight)
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:36]:
```suggestion
	logger.Info("Ending benchmark period", "height", endHeight)
```

File: test/e2e/runner/benchmark.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Benchmark is a simple function for fetching, calculating and printing
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:13]: Might be helpful to describe what `benchmarkLength` is in the godoc.

File: test/e2e/runner/benchmark.go [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+		t.to,
+		t.mean,
+		t.std,
+		t.max,
+		t.min,
+	)
+}
+
+// fetchBlockChain fetches a complete, up-to-date block history from
+// the freshest testnet archive node.
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:81]: what's "freshest"?

File: test/e2e/runner/benchmark.go [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+	ctx := context.Background()
+	s, err := c.Status(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	to := s.SyncInfo.LatestBlockHeight
+	from := to - benchmarkLength + 1
+	if from <= testnet.InitialHeight {
+		return nil, fmt.Errorf("tesnet was unable to reach required height for benchmarking. (latest height %d)", to)
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:103]:
```suggestion
		return nil, fmt.Errorf("tesnet was unable to reach required height for benchmarking (latest height %d)", to)
```

File: test/e2e/runner/benchmark.go [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+		intervals[i-1] = block.Header.Time.Sub(lastTime)
+		lastTime = block.Header.Time
+	}
+	return intervals
+}
+
+func extractTestnetStats(intervals []time.Duration) testnetStats {
+	var sum, mean, max, min time.Duration
+	var std float64
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:148]: nit: group vars

**Review by alexanderbez — COMMENTED** [review]
Amazing 👏

File: test/e2e/runner/benchmark.go (hunk 6 above)
Comment (cmwaters) [reply test/e2e/runner/benchmark.go:81]: Sorry this was just directly copied from another part of the e2e tests. Will reword it. It basically means fetch from the first archive node

File: test/e2e/runner/benchmark.go [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+	startHeight  int64
+	endHeight   int64
+
+	// average time to produce a block
+	mean time.Duration
+	// standard deviation of block production
+	std  float64
+	// longest time to produce a block 
+	max  time.Duration
+	// shortest tiem to produce a block
```
Comment (alexanderbez) [inline test/e2e/runner/benchmark.go:67]:
```suggestion
	// shortest time to produce a block
```

**Review by alexanderbez — APPROVED** [review]

File: test/e2e/runner/benchmark.go (hunk 9 above)
Comment (alexanderbez) [reply test/e2e/runner/benchmark.go:67]: Along with a few other linting errors.

**Pull request merged**